### ISSUE TITLE
uncaughtException from a bad createClient parameter

### ISSUE THREAD

**Issue by mjhm** [ldapjs/node-ldapjs#418]
This script generates an `uncaughtException`
```js
var ldap = require('ldapjs');

process.on('uncaughtException', function () {
  console.log("I really don't want to be here");
})


try {
  client = ldap.createClient({url: 'ldap://localhost:13891389'});
  client.on('error', function (err) {
    console.log('I wish that there was an error event for this', err);
  });
} catch (err) {
  console.log('or, even better, I wish this error was catchable', err);
}


setTimeout(function () { console.log('exiting')}, 1000);
```

**Comment by jsumners** [comment]
The issue is in the Node.js URL parser recognizing that `13891389` is an invalid port and throwing an error because of it. But this library's internal URL parser was not taking into account that the Node parser could throw an error, so it wasn't bubbling up the error correctly.

**Comment by jsumners** [comment]
This issue should be resolved in `ldapjs@2.0.0-pre.3`. Please give it a try if you are still affected.

Note: you'll have to wrap the `createClient` in a `try/catch`. The error occurs too early for the event emitter to handle the error.

**Comment by jsumners** [comment]
⚠️ This issue has been locked due to age. If you have encountered a recent
problem that seems to be covered by this issue, please open a new issue.

Please include a [minimal reproducible example](https://en.wikipedia.org/wiki/Minimal_reproducible_example)
when opening a new issue.
